src/Stratis.Bitcoin.IntegrationTests/RPC/RpcBitcoinMutableTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Threading;
+using NBitcoin;
+using NBitcoin.RPC;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Stratis.Bitcoin.IntegrationTests.RPC
+{
+    /// <summary>
+    /// These tests are for RPC tests that require modifying the chain/nodes. 
+    /// Setup of the chain or nodes can be done in each test.
+    /// </summary>
+    public class RpcBitcoinMutableTests
+    {
+        /// <summary>
+        /// <seealso cref="https://github.com/MetacoSA/NBitcoin/blob/master/NBitcoin.Tests/RPCClientTests.cs">NBitcoin test CanGetRawMemPool</seealso>
+        /// </summary>
+        [Fact]
+        public void GetRawMemPoolWithValidTxThenReturnsSameTx()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create())
+            {
+                CoreNode node = builder.CreateNode();
+                builder.StartAll();
+
+                RPCClient rpcClient = node.CreateRPCClient();
+
+                // generate 101 blocks
+                node.GenerateAsync(101).GetAwaiter().GetResult();
+
+                uint256 txid = rpcClient.SendToAddress(new Key().PubKey.GetAddress(rpcClient.Network), Money.Coins(1.0m), "hello", "world");
+                uint256[] ids = rpcClient.GetRawMempool();
+                Assert.Single(ids);
+                Assert.Equal(txid, ids[0]);
+            }
+        }
+
+        /// <summary>
+        /// <seealso cref="https://github.com/MetacoSA/NBitcoin/blob/master/NBitcoin.Tests/RPCClientTests.cs">NBitcoin test CanAddNodes</seealso>
+        /// </summary>
+        [Fact]
+        public void AddNodeWithValidNodeThenExecutesSuccessfully()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create())
+            {
+                CoreNode nodeA = builder.CreateNode();
+                CoreNode nodeB = builder.CreateNode();
+                builder.StartAll();
+                RPCClient rpc = nodeA.CreateRPCClient();
+                rpc.RemoveNode(nodeA.Endpoint);
+                rpc.AddNode(nodeB.Endpoint);
+
+                AddedNodeInfo[] info = null;
+                WaitAssert(() =>
```

Comment:
I wonder whether I might be able to use this TestHelper.WaitLoop() instead?

Reply:
Ah yes @mikedennis totally. Use the helper ;)